File: docs/guides/ruby/deleteMultipleIndices.rb
require "algolia"

begin
  # You need an API key with `deleteIndex`
  client = Algolia::SearchClient.create("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")

  # List all indices
  indices = client.list_indices

  # Primary indices don't have a `primary` key
  primary_indices, replica_indices = indices.items.partition { |index| index.primary.nil? }.map(&:to_a)

  # Delete primary indices first
  if primary_indices.any?
    requests = primary_indices.map { |index|
      Algolia::Search::BatchRequest.new(action: Algolia::Search::Action::DELETE, indexName: index.name)
    }
    client.multiple_batch(Algolia::Search::BatchParams.new(requests: requests))
    print("Deleted primary indices.")
  end

  # Now, delete replica indices
  if replica_indices.any?
    requests = replica_indices.map { |index|
      Algolia::Search::BatchRequest.new(action: Algolia::Search::Action::DELETE, indexName: index.name)
    }
    client.multiple_batch(Algolia::Search::BatchParams.new(requests: requests))
    print("Deleted replica indices.\n")
  end

rescue Exception => e
  puts("An error occurred: #{e}")
end

File: docs/guides/scala/src/main/scala/deleteMultipleIndices.scala
import scala.concurrent.ExecutionContext.Implicits.global
import scala.concurrent.Future

import algoliasearch.api.SearchClient
import algoliasearch.config.*
import algoliasearch.extension.SearchClientExtensions
import algoliasearch.search.{Action, BatchParams, MultipleBatchRequest}

def deleteMultipleIndices(): Future[Unit] = {
  // You need an API key with `deleteIndex`
  val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")

  client
    .listIndices(
    )
    .flatMap { indices =>
      val (primaryIndices, replicaIndices) = indices.items.partition(_.primary.isEmpty)

      val deletePrimary = if (primaryIndices.nonEmpty) {
        val requests = primaryIndices.map(index => MultipleBatchRequest(Action.Delete, None, index.name))
        client
          .multipleBatch(
            batchParams = BatchParams(
              requests = requests
            )
          )
          .map { _ =>
            println("Deleted primary indices.")
          }
      } else Future.unit

      val deleteReplica = if (replicaIndices.nonEmpty) {
        val requests = replicaIndices.map(index => MultipleBatchRequest(Action.Delete, None, index.name))
        client
          .multipleBatch(
            batchParams = BatchParams(
              requests = requests
            )
          )
          .map { _ =>
            println("Deleted replica indices.")
          }
      } else Future.unit

      for {
        _ <- deletePrimary
        _ <- deleteReplica
      } yield ()
    }
    .recover { case ex: Exception =>
      println(s"An error occurred: ${ex.getMessage}")
    }
}
